test/metrics/metrics_test.go

```diff
@@ -0,0 +1,735 @@
+				RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+				Execute().Resources()
+
+			// Create the BannedUser resource
+			bannedUser := banUser(t, hostAwait, userSignup.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey])
+
+			// Confirm the user is banned
+			_, err := hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*15)).WaitForUserSignup(userSignup.Name,
+				wait.UntilUserSignupHasConditions(ConditionSet(Default(), ApprovedByAdmin(), Banned())...))
+			require.NoError(t, err)
+
+			t.Run("verify metrics", func(t *testing.T) {
+				metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 1)
+				metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 1)
+				metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 1)
+				metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")
+				metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)  // no user on member1
+				metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+			})
+
+			t.Run("unban the banned user", func(t *testing.T) {
+				// Get metrics assertion helper for testing metrics
+				metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName, member2Await.ClusterName})
+
+				// Unban the user
+				err = hostAwait.Client.Delete(context.TODO(), bannedUser)
+				require.NoError(t, err)
+
+				// Confirm the BannedUser is deleted
+				err = hostAwait.WaitUntilBannedUserDeleted(bannedUser.Name)
+				require.NoError(t, err)
+
+				// Confirm the user is provisioned
+				userSignup, err = hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*10)).WaitForUserSignup(userSignup.Name,
+					wait.UntilUserSignupHasConditions(ConditionSet(Default(), ApprovedByAdmin())...),
+					wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValueApproved))
+				require.NoError(t, err)
+
+				// Confirm the MUR is created
+				_, err = hostAwait.WaitForMasterUserRecord(mur.Name)
+				require.NoError(t, err)
+
+				t.Run("verify metrics", func(t *testing.T) {
+					metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 0)         // unchanged: user signup already existed
+					metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 1) // user approved
+					metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 0)   // unchanged: banneduser already existed
+					metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 1, "domain", "external")
+					metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 1, "cluster_name", memberAwait.ClusterName)  // user provisioned on member1
+					metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+				})
+			})
+		})
+
+		t.Run("manually created usersignup with preexisting banneduser", func(t *testing.T) {
```

Comment:
this is also more likely a test of the UserSignup controller logic, and not of the metrics